folly/Traits.h

```diff
@@ -509,15 +509,8 @@ struct IsRelocatable< std::pair<T, U> >
         IsRelocatable<U>::value> {};
 
 // Is T one of T1, T2, ..., Tn?
-template <class T, class... Ts>
-struct IsOneOf {
-  enum { value = false };
-};
-
-template <class T, class T1, class... Ts>
-struct IsOneOf<T, T1, Ts...> {
-  enum { value = std::is_same<T, T1>::value || IsOneOf<T, Ts...>::value };
-};
+template <typename T, typename ...Ts>
+using IsOneOf = Negation<StrictConjunction<Negation<std::is_same<T, Ts>>...>>;
```

Comment:
Add `StrictDisjunction`, which should probably exist anyway, and then have `IsOneOf` be defined in terms of that?